### PR TITLE
Refactor Attachment#push_event_data

### PR DESCRIPTION
**Pull request by arku**
Fixes https://github.com/chatwoot/chatwoot/issues/53

### CODE REVIEW

File: app/models/attachment.rb [hunk 1]
```diff
@@ -9,36 +9,47 @@ class Attachment < ApplicationRecord
...
-    data = {
+    return base_data.merge(location_data) if file_type.to_sym == :location
+    return base_data.merge(fallback_data) if file_type.to_sym == :fallback
+
+    base_data.merge(file_data)
+  end
+
+  private
+
+  def file_data
```
Comment (arku) [inline app/models/attachment.rb]: @pranavrajs Couldn't come up with a better name for this method. What should we call this?

**Comment by pranavrajs** [comment]
@subintp / @sojan-official Can you please take a look at this PR?

File: app/models/attachment.rb (hunk 1 above)
Comment (sojan-official) [reply app/models/attachment.rb]: `file_meta_data` ?

File: app/models/attachment.rb [hunk 2]
```diff
@@ -9,36 +9,47 @@ class Attachment < ApplicationRecord
...
-    elsif :fallback == file_type.to_sym
-      data.merge!({
-        fallback_title: fallback_title,
-        data_url: external_url
-      })
-   end
-   data
   end
 
-  private
```
Comment (sojan-official) [inline app/models/attachment.rb]: why did we remove the private?

Comment (arku) [reply app/models/attachment.rb]: @sojan-official Please see https://github.com/chatwoot/chatwoot/pull/93/files#diff-7b0581da1680146a96496be759fc3240R18

File: app/models/attachment.rb (hunk 1 above)
Comment (arku) [reply app/models/attachment.rb]: Let me update it.

**Comment by arku** [comment]
@sojan-official I have addressed your comments. Do you mind taking another look?

Thanks!

**Pull request merged**